app/javascript/packages/components/text-input.spec.tsx

```diff
@@ -45,4 +45,13 @@ describe('TextInput', () => {
 
     expect(ref.current).to.be.an.instanceOf(HTMLInputElement);
   });
+
+  it('renders with a hint', () => {
+    const { getByText } = render(<TextInput label="Input" hint="Something special" />);
+
+    const input = getByText('Something special');
+
+    expect(input).to.be.an.instanceOf(HTMLDivElement);
+    expect(input.classList.contains('usa-hint')).to.be.true();
```

Comment:
The variable name doesn't feel accurate here, since it's not an input we're testing against, it's the hint.

In other cases, we've used `computeAccessibleDescription` to make sure that the input's accessible description includes the expected hint, to ensure that the accessible relationship is working correctly.

e.g.

```ts
import { computeAccessibleDescription } from 'dom-accessibility-api';
```

```suggestion
    const { getByLabel } = render(<TextInput label="Input" hint="Something special" />);

    const input = getByLabel('Input');
    const description = computeAccessibleDescription(input);

    expect(description).to.equal('Something special');
```

Then again, I like your idea of asserting that the expected USWDS class is also applied, so maybe we could do a combination of both approaches.